lib/enum_state_machine/integrations/active_record.rb

```diff
@@ -455,10 +455,13 @@ def define_static_state_initializer
           # attributes passed into #initialize
           define_helper :class, <<-end_eval, __FILE__, __LINE__ + 1
             def column_defaults(*) #:nodoc:
-              result = super
+              super
+              @column_defaults = _default_attributes.dup.to_hash
```

Comment:
This is what's returned from:
- Rails 4.2.11.3: https://github.com/rails/rails/blob/v4.2.11.3/activerecord/lib/active_record/model_schema.rb#L250
- Rails 5.2.6: https://github.com/rails/rails/blob/v5.2.6/activerecord/lib/active_record/model_schema.rb#L378
- Rails 6.1.4.4: https://github.com/rails/rails/blob/v6.1.4.4/activerecord/lib/active_record/model_schema.rb#L456
- Rails 7.0.1: https://github.com/rails/rails/blob/v7.0.1/activerecord/lib/active_record/model_schema.rb#L471

**However**, Rails 4.1 and lower does not define `_default_attributes`, so the `gemspec` will need to be modified.
- Rails 4.1.16: https://github.com/rails/rails/blob/v4.1.16/activerecord/lib/active_record/model_schema.rb#L255

Reply:
The gemspec has
```ruby
  s.add_dependency "rails", "> 4.0", "< 7.0"
```
so based on my research we could get away with `rails > 4.2`. If you want to drop support for Rails 4 altogether, I'm fine with that.

Reply:
I vote to drop any Rails < 5.2.